packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  Reference,
+  Variable,
+} from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+
+export = {
+  name: "load-object-before-read",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"problem",
+    messages: {
+      loadBeforeRead: "An explicit load call on '{{name}}' for '{{loadValue}}' needs to be made before reading a proxy object",
+    },
+    docs: {
+      description: 
+        "Before you can read the properties of a proxy object, you must explicitly load the properties",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Possible Errors",
+      recommended: <false | "error" | "warn">false,
+      url: 
+        "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model?view=powerpoint-js-1.1#load",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    function getValueThatHadToBeLoaded(referenceNode: Reference): string | undefined {
+      if(referenceNode.identifier.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && referenceNode.identifier.parent?.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+        return referenceNode.identifier.parent.property.name;
+      }
+      return undefined;
+    }
+
+    function isLoadFunction(reference: Reference): boolean {
+      if(reference.identifier.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && reference.identifier.parent.property.type === TSESTree.AST_NODE_TYPES.Identifier
+        && reference.identifier.parent.property.name === "load") {
+        return true;
+      }
+      return false;
+    }
+
+    function wasCreatedByGetFunction(referenceNode: Reference): boolean {
+      const variable = referenceNode.resolved;
+      let getFunctionFound = false;
+      variable?.references.forEach((reference: Reference) => {
+          if(reference.identifier.parent?.type === TSESTree.AST_NODE_TYPES.VariableDeclarator
+            && reference.identifier.parent.init?.type === TSESTree.AST_NODE_TYPES.CallExpression
+            && reference.identifier.parent.init.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+            && reference.identifier.parent.init.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+              const functionName = reference.identifier.parent.init.callee.property.name;
+              if(functionName === "getSelectedRange"
+                || functionName === "getItem" 
+                || functionName === "getRange") {
+                getFunctionFound = true;
+              }
+          }
+        });
+      return getFunctionFound;
+    }
+
+    function isLoaded(referenceNode: Reference): boolean {
+      const variable = referenceNode.resolved;
+      let loadFound = false;
+      const valueRead = getValueThatHadToBeLoaded(referenceNode);
+
+      variable?.references.forEach((reference: Reference) => {
+        if(reference === referenceNode) return;
```

Comment:
Aren't you supposed to return a boolean?

Reply:
This is not really a return as normal. This is acting as a `continue` statement. 
I think this is a confusing thing in Javascript, but it is explained here:
[Return in ForEach](https://stackoverflow.com/questions/22585032/does-the-foreach-loop-allow-using-break-and-continue#:~:text=As%20already%20answered%2C%20you%20cannot%20use%20continue%20or,continue%20or%20throw%20new%20Error%20%28%29%20to%20break.)